src/table/hooks/use-view-service.ts

```diff
@@ -1,16 +1,21 @@
 import { useMemo } from '@nebula.js/stardust';
-import type { ViewService } from '../../types';
+import type { SnapshotData, ViewService, ViewState } from '../../types';
 
-const createViewService = (): ViewService => {
+const createViewService = (viewState: ViewState, snapshotData?: SnapshotData): ViewService => {
   return {
     qLeft: 0,
     qTop: 0,
     qWidth: 0,
     qHeight: 0,
-    scrollLeft: 0,
+    visibleTop: viewState?.visibleTop,
+    visibleHeight: viewState?.visibleHeight,
+    scrollLeft: snapshotData?.content?.scrollLeft ?? (viewState?.scrollLeft || 0),
```

Comment:
would be nice with consistency here, does 
```suggestion
    scrollLeft: snapshotData?.content?.scrollLeft ?? viewState?.scrollLeft ?? 0,
```
work here?